src/frontend/admin/src/components/templates/users/inputs/search/UserSearch.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from "react";
+import { useState } from "react";
+import {
+  RHFAutocompleteSearchProps,
+  RHFSearch,
+} from "@/components/presentational/hook-form/RHFSearch";
+import { Maybe } from "@/types/utils";
+import { User } from "@/services/api/models/User";
+import { useUsers } from "@/hooks/useUsers/useUsers";
+
+export function UserSearch(props: RHFAutocompleteSearchProps<User>) {
+  const [query, setQuery] = useState("");
+  const users = useUsers({ query });
+
+  return (
+    <RHFSearch
+      {...props}
+      items={users.items}
+      loading={users.states.fetching}
+      onFilter={(term) => setQuery(term)}
+      getOptionLabel={(option: Maybe<User>) => option?.username ?? ""}
+      isOptionEqualToValue={(option, value) =>
+        option.username === value.username
+      }
+    />
+  );
```

Comment:
Currently this search field does not allow to filter user through email, firstname or lastname while api endpoint allow that. I feel you should write test for those kind of component to ensure all scenarii are working (search by username, firstname, lastname or email)